tests/Exizent.ComponentModel.DataAnnotations.Tests/AvailableIfContainsAttributeTests.cs

```diff
@@ -0,0 +1,250 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Xunit;
+
+namespace Exizent.ComponentModel.DataAnnotations.Tests;
+
+public class AvailableIfContainsAttributeTests
+{
+    public enum TestEnum
+    {
+        Value1,
+        Value2,
+        Value3,
+        Value4
+    }
+    
+    class InvalidDependentPropertyTestClass
+    {
+        [AvailableIfContains("PropertyThatDoesNotExist", 1)]
+        public string? Value { get; set; }
+    }
+    
+    class InvalidDependentPropertyTypeTestClass
+    {
+        public int DependentProperty { get; set; }
+        [AvailableIfContains(nameof(DependentProperty), 1)]
+        public string? Value { get; set; }
+    }
+    
+    [Fact]
+    public void ShouldThrowInvalidOperationExceptionForInvalidDependentProperty()
+    {
+        var model = new InvalidDependentPropertyTestClass
+        {
+            Value = Guid.NewGuid().ToString()
+        };
+
+        Action action = () => ValidateModel(model);
+
+        action.Should()
+            .Throw<InvalidOperationException>()
+            .WithMessage("The dependent property '*' does not exist");
+    }
+    
+    [Fact]
+    public void ShouldThrowInvalidOperationExceptionForInvalidDependentPropertyType()
+    {
+        var model = new InvalidDependentPropertyTypeTestClass
+        {
+            DependentProperty = 4,
+            Value = Guid.NewGuid().ToString()
+        };
+
+        Action action = () => ValidateModel(model);
+
+        action.Should()
+            .Throw<InvalidOperationException>()
+            .WithMessage("The dependent property '*' must be of type IEnumerable");
+    }
+    
+
+    public class SinglePossibleDependentValueTests
```

Comment:
split this to its own file?

Reply:
It shares stuff with the outer class at the moment.

Reply:
it would be easier to understand usage by reading the happy paths first